build.gradle

```diff
@@ -523,30 +521,32 @@ task coverage_cpp_report {
         mkdir "${buildDir}/reports/cpp"
     }
     doLast {
-        exec {
-            workingDir "${buildDir}/cmake-coverage"
-            commandLine 'lcov', '--capture', '--directory', "${buildDir}/cmake-coverage/CMakeFiles/amazonCorrettoCryptoProvider.dir/csrc", '--output-file', "${buildDir}/cmake-coverage/coverage.info", '--rc', 'lcov_branch_coverage=1', '--ignore-errors', lcovIgnore
-        }
-        // Convert absolute to relative paths
-        exec {
-            workingDir "${buildDir}/cmake-coverage"
-            commandLine 'perl', '-i', '-pe', "s[^SF:${projectDir}/*][SF:]", "${buildDir}/cmake-coverage/coverage.info"
-        }
-        // lcov captures coverage data for inline functions in system headers; strip this out to avoid
-        // polluting our metrics with unused STL code.
-        exec {
-            workingDir "${buildDir}/cmake-coverage"
-            commandLine 'lcov', '-e', "${buildDir}/cmake-coverage/coverage.info", 'csrc/*', '--rc', 'lcov_branch_coverage=1', '--ignore-errors', lcovIgnore
-            standardOutput = new FileOutputStream("${buildDir}/reports/cpp/coverage.info")
-        }
-        exec {
-            workingDir "${buildDir}/cmake-coverage"
-            commandLine 'gcovr', '-r', "${projectDir}", '--xml'
-            standardOutput = new FileOutputStream("${buildDir}/reports/cpp/cobertura.xml")
-        }
-        exec {
-             workingDir projectDir
-             commandLine 'genhtml', '-o', "${buildDir}/reports/cpp", '--rc', 'genhtml_branch_coverage=1', "${buildDir}/reports/cpp/coverage.info", '--ignore-errors', lcovIgnore
+        if (!isLegacyBuild) {
```

Comment:
why does linkage affect our coverage collection and reporting logic?

also, why do we now condition this on `isLegacyBuild` where we didn't before?

Reply:
As part of this change, we started running coverage during legacy build. Legacy build has old versions of lcov and gcov, which are not very reliable. Since we already perform coverage in other CI tasks, we don't need to do it for legacy build.
Having said that, this change is reverted (including adding `--gcov-ignore-parse-errors`) and instead for regular builds we add `-fprofile-update=atomic` compile flag when computing coverage.